src/Logic/UsersHelper.php

```diff
@@ -67,36 +69,78 @@ public static function get_user( array $data ): WP_User|bool {
 	}
 
 	/**
-	 * Get a username that is not in use from a desired username.
+	 * Confirms that the given username (`wp_users`.`user_login`) is unused.
+	 *
+	 * @param string $username The username (`wp_users`.`user_login`) to check.
+	 * @param int    $exclude_user_id A user ID to exclude from the check.
+	 *
+	 * @return bool
+	 */
+	public static function is_username_unused( string $username, int $exclude_user_id = 0 ): bool {
+		// Check to see if raw $username is unused. Assumption is that all sanitation (if necessary under the context) has already been performed on $username.
+		// `get_user_by` is not used here because it does some sanitization (via `sanitize_user()`), as well as caching.
+		global $wpdb;
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users -- We need raw uncached query results.
+		$prepared_sql = $wpdb->prepare( "SELECT ID FROM $wpdb->users WHERE user_login = %s", $username );
+
+		if ( $exclude_user_id ) {
+			// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: $sql_prepared is a prepared statement.
+			$prepared_sql = $wpdb->prepare( "$prepared_sql AND ID <> %d", $exclude_user_id );
+		}
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.NotPrepared
+		$found_no_match = null === $wpdb->get_var( $prepared_sql );
+
+		if ( $found_no_match && has_filter( 'nmt_additional_unused_username_check' ) ) {
```

Comment:
There shouldn't be any reason to check if the filter exists. `apply_filters` does that already. The filtered value needs to be passed to the filter though: `apply_filters( 'nmt_additional_unused_username_check', $found_no_match, $username, $exclude_user_id );` (maybe rename the var to $exists). 

Also – if we can add a docblock over the filter apply it's easier for people to see the intention.